### PR TITLE
Remove MetricProcessor from the SDK spec (for now)

### PR DESCRIPTION
**Pull request by reyang**
## Changes

Discussed during the [7/29/2021 Metrics SIG](https://docs.google.com/document/d/1-bCYkN-DWJq4jw1ybaDZYYmx-WAe6HnwfWbkm8d57v8/edit#). We're removing the `MetricProcessor` for now, most likely we will not expose such interface in the first release of the metrics SDK specification. If we need something similar, we will cover it in a separate PR (probably together with the exporters).

@jsuereth FYI

### CODE REVIEW

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -43,10 +42,10 @@ to create an
...
-[`Views`](#view)) MUST be managed solely by the `MeterProvider` and the SDK MUST
-provide a way to configure all options that are implemented by the SDK. This MAY be done at the
-time of MeterProvider creation if appropriate.
+[MetricExporters](#metricexporter) and [`Views`](#view)) MUST be managed solely
+by the `MeterProvider` and the SDK MUST provide a way to configure all options
+that are implemented by the SDK. This MAY be done at the time of MeterProvider
+creation if appropriate.
 
 The `MeterProvider` MAY provide methods to update the configuration. If
 configuration is updated (e.g., adding a `MetricProcessor`), the updated
```
Comment (cijothomas) [inline specification/metrics/sdk.md:51]:
```suggestion
configuration is updated (e.g., adding a `MeasurementProcessor`), the updated
```

**Review by cijothomas — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 2]
```diff
@@ -43,10 +42,10 @@ to create an
 instance which is stored on the created `Meter`.
 
 Configuration (i.e., [MeasurementProcessors](#measurementprocessor),
```
Comment (jsuereth) [inline specification/metrics/sdk.md:44]: Nit: I think we're not exposing MeasurementProcessor in the first version either, right?  Just Views + MetricExporters.

**Review by jsuereth — APPROVED** [review]

File: specification/metrics/sdk.md (hunk 2 above)
Comment (reyang) [reply specification/metrics/sdk.md:44]: I think we need something that is generic (either we call it a processor or reader or something else), which allows exporters to be hooked up. I'd like to cover it in a separate PR.
Please refer to [this comment](https://github.com/open-telemetry/opentelemetry-specification/pull/1730#discussion_r669024476) from @bogdandrutu.

Comment (reyang) [reply specification/metrics/sdk.md:44]: Ping back from https://github.com/open-telemetry/opentelemetry-specification/issues/617#issuecomment-856913098.

**Review by jmacd — APPROVED** [review]

**Pull request merged**